openedx/core/djangoapps/discussions/transformers.py

```diff
@@ -0,0 +1,46 @@
+"""
+Discussions Topic Link Transformer
+"""
+
+from openedx.core.djangoapps.content.block_structure.transformer import BlockStructureTransformer
+from openedx.core.djangoapps.discussions.models import DiscussionTopicLink, DiscussionsConfiguration
+
+
+class DiscussionsTopicLinkTransformer(BlockStructureTransformer):
+    """
+    A transformer that adds discussion topic context to the xblock.
+    """
+    WRITE_VERSION = 1
+    READ_VERSION = 1
+    EXTERNAL_ID = "discussions_external_id"
+    EMBED_URL = "discussions_embed_url"
+
+    @classmethod
+    def name(cls):
+        """
+        Unique identifier for the transformer's class;
+        same identifier used in setup.py.
+        """
+        return "discussions_link"
+
+    def transform(self, usage_info, block_structure):
+        """
+        loads override data into blocks
+        """
+        provider_type = DiscussionsConfiguration.get(usage_info.course_key).provider_type
+        topic_links = DiscussionTopicLink.objects.filter(
+            context_key=usage_info.course_key,
+            provider_id=provider_type,
+            enabled_in_context=True,
+        )
+        for topic_link in topic_links:
+            block_structure.override_xblock_field(
+                topic_link.usage_key,
+                "discussions_external_id",
+                topic_link.external_id,
+            )
+            block_structure.override_xblock_field(
+                topic_link.usage_key,
+                "discussions_embed_url",
+                f"http://localhost:2000/discusions/{usage_info.course_key}/topics/{topic_link.external_id}"
```

Comment:
TODO: Get this from the setting, and check for the setting before running the transformer.

Reply:
This is added in https://github.com/edx/edx-platform/pull/29285 so once that PR is merged that change can be added.